tests/lib/nested-assets/dbx/README.md

```diff
@@ -0,0 +1,142 @@
+# Files list
+
+`OVMF_VARS.test.fd` custom OVMF variables block content with enrolled PK, KEK
+and Snakeoil keys. Used for dbx testing.
+
+Matches the following code block:
+
+```
+2b372dd411e38931fb12e80834cbb7ff54b3d36b6c2c63629ab1a569a1d87f1b  OVMF_CODE.secboot.fd
+```
+
+# Generate a new data set
+
+``` sh
+
+# grab snakeoil keys
+cp PkKek-1-snakeoil.pem PkKek-1-snakeoil.crt
+
+make clean
+make clean-disk
+
+# generate data
+# signing keys: sign.crt and the snakeoil key
+# initial dbx blacklists bogus.crt, and an update which blacklists other-bogus.crt
+make dataset SIGNING_KEYS='sign.crt PkKek-1-snakeoil.crt' BLACKLIST_KEYS='bogus.crt'
+
+# prepare disk for enrolling keys
+make tree/db.auth
+make tree/dbx-blacklist.auth
+make disk.img
+```
+
+
+# QEMU
+
+``` sh
+cp OVMF_VARS.fd OVMF_VARS.test.fd
+
+# run without snapshot mode
+qemu-system-x86_64 -enable-kvm -smp 4 -m 2048 -cpu host -machine q35 -global ICH9-LPC.disable_s3=1 \
+        -drive file=./OVMF_CODE.secboot.fd,if=pflash,format=raw,unit=0,readonly=on \
+        -drive file=./OVMF_VARS.test.fd,if=pflash,format=raw \
+        -drive file=efi-auth/disk.img,if=none,id=disk1,snapshot=on \
+        -device virtio-blk-pci,drive=disk1,bootindex=1 
+        
+# device configuration -> secure boot -> custom
+# 1. enroll PK
+# 2. enroll KEK
+# 3. enroll DB
+# 4. enroll DBX?
+
+# reset
+# pwoeroff
+```
```

Comment:
If we build [efitools](https://git.kernel.org/pub/scm/linux/kernel/git/jejb/efitools.git), we can create `LockDown.efi`. Then we can create an `startup.nsh` script that calls it then shutdown. We just need OVMF with a shell.